src/etc/init.d/S50dropbear

```diff
@@ -15,9 +15,15 @@ start() {
 	umask 077
 	local dropbear_key_dir="/etc/dropbear"
 
-	# Handle symlinked directories
-	if [ -L "$dropbear_key_dir" ]; then
-		dropbear_key_dir=$(readlink -f "$dropbear_key_dir")
+	# If /etc/dropbear is not a directory, and
+	#   - the filesystem is RO (i.e. we can not rm the symlink),
+	#     create the directory pointed to by the symlink.
+	#   - the filesystem is RW (i.e. we can rm the symlink),
+	#     replace the symlink with an actual directory
+	if ! [ -d /etc/dropbear ]; then
+		if ! rm -f /etc/dropbear; then
+			dropbear_key_dir=$(readlink "$dropbear_key_dir")
+		fi
```

Comment:
shouldn't /etc/dropbear above be replaced by `$dropbear_key_dir` ? just to avoid repeating oneself.

For reference, I'm pretty sure you took the logic from
https://github.com/analogdevicesinc/buildroot/blob/master/package/dropbear/dropbear.service